Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -739,6 +749,10 @@ extension MergedSQLiteBookmarks: SyncableBookmarks {
             }
         }
     }
+
+    public func getServerRecordsCount() -> Deferred<Maybe<Int>> {
+        return self.local.getServerRecordsCount()
```

Comment:
Abstractions leaking everywhere!

I suggest you put this next to the implementation of `synchronousBufferCount` in `BookmarksBufferStorage`, not in `Merged…`.